test/hooks.js

```diff
@@ -15,8 +15,18 @@
 /* global beforeEach afterEach */
 'use strict';
 
-const cloudant = require('@cloudant/cloudant')(
-  (process.env.COUCH_BACKEND_URL) ? { url: process.env.COUCH_BACKEND_URL } : 'https://no-couch-backend-url-set.test');
+const { CloudantV1 } = require('@ibm-cloud/cloudant');
+const url = new URL((process.env.COUCH_BACKEND_URL) ? process.env.COUCH_BACKEND_URL : 'https://no-couch-backend-url-set.test');
+const { BasicAuthenticator, NoAuthAuthenticator } = require('ibm-cloud-sdk-core');
+const authenticator = (url.username) ? new BasicAuthenticator({ username: url.username, password: url.password }) : new NoAuthAuthenticator();
```

Comment:
Handling missing username/password would be a nice to have.

Reply:
I'm not sure I follow - if the `username` is missing we use `NoAuthAuthenticator` - which is a valid use case for our mocks (which don't mock the session handling aspects) or indeed if one was testing against e.g. an admin party CouchDB.
In the case that the `username` is supplied but not the `password` then if a password is required by the server the test will fail anyway wtih a `401`.
Do you mean to make an assumption that a `password` is always needed if a `username` is supplied and assert it separately?

Reply:
> In the case that the `username` is supplied but not the `password` then if a password is required by the server the test will fail anyway wtih a `401`.
> Do you mean to make an assumption that a `password` is always needed if a `username` is supplied and assert it separately?

Yeah, but not necessary, because it will fail on the server-side (that's why I wrote _a nice to have_).